dsp/modules/llama.py

```diff
@@ -0,0 +1,96 @@
+from attr import dataclass
+
+try:
+    from llama_cpp import Llama
+except ImportError as exc:
+    raise ModuleNotFoundError(
+        "You need to install llama_cpp library to use gguf models."
+    ) from exc
+
+from dsp.modules.lm import LM
+from typing import Any, Literal, Optional
+import os
+
+
+class LlamaCpp(LM):
+    def __init__(
+        self,
+        model: str,
+        llama_model: Llama,
+        model_type: Literal["chat", "text"] = None,
+        **kwargs,
+    ):
+        super().__init__(model)
+
+        default_model_type = "chat" if "instruct" not in model else "text"
```

Comment:
does this make sense here? this is for OpenAI models

Reply:
i will remove that to